cmake/alpakaCommon.cmake

```diff
@@ -14,6 +14,14 @@ include(CMakePrintHelpers) # for easier printing of variables and properties
 #-------------------------------------------------------------------------------
 # Options.
 
+# CMake build option
+set(alpaka_BUILD_TYPE "Release;RelWithDebInfo;Debug")
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the build type for the project" FORCE)
+endif()
```

Comment:
Does that mean that if I `find_package(alpaka)` from a LLAMA example, which does not specify a `CMAKE_BUILD_TYPE`, alpaka will set the build type of my LLAMA example? Because that might be invasive.

Reply:
If LLAMA is setting something else as buildtype and you define your own build types your app setting will not be overwritten.
If your app is keeping CMake in an undefined build type it will be set by alpaka.

Reply:
LLAMA is not setting anything, because the build type is up to the user.
> If your app is keeping CMake in an undefined build type it will be set by alpaka.
So alpaka is injecting a build type now, depending on whether the LLAMA examples find it or not. I find that dangerous.

I would like to propose to set the build type only for alpaka tests and examples and not when the library is included.

Reply:
> I would like to propose to set the build type only for alpaka tests and examples and not when the library is included.

We can but this means that your application is not compiled with a defined CMake build type.

@j-stephan Do you also think we should keep the undefined behavior?

Reply:
Well, the CMake developers don't seem opposed to having a default type: https://blog.kitware.com/cmake-and-the-default-build-type/

We just need to make sure that this applies to other generators (MSVC...), too that don't obey `CMAKE_BUILD_TYPE`.

Reply:
> We can but this means that your application is not compiled with a defined CMake build type.
That is true and it is usually bad. But I think alpaka should not fix this for downstream projects, because it is surprising.

However, if you truly want this then go ahead. I don't mind much for LLAMA. Let's see if someone else complains later.

Maybe the new `CMakePresets.txt` could also be used to address this.